modules/openapi-generator/src/main/resources/C-libcurl/CMakeLists.txt.mustache

```diff
@@ -86,14 +87,66 @@ include(PreTarget.cmake OPTIONAL)
 add_library(${pkgName} ${SRCS} ${HDRS})
 # Link dependent libraries
 if(NOT CMAKE_VERSION VERSION_LESS 3.4)
-    target_link_libraries(${pkgName} OpenSSL::SSL OpenSSL::Crypto)
+    target_link_libraries(${pkgName} PRIVATE OpenSSL::SSL OpenSSL::Crypto)
 endif()
-target_link_libraries(${pkgName} ${CURL_LIBRARIES} )
+target_link_libraries(${pkgName} PUBLIC ${CURL_LIBRARIES} )
+target_include_directories(
+    ${pkgName} PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+    $<INSTALL_INTERFACE:include>
+)
 
 include(PostTarget.cmake OPTIONAL)
 
-#install library to destination
-install(TARGETS ${pkgName} DESTINATION ${CMAKE_INSTALL_PREFIX})
+# installation of libraries, headers, and config files
+if (NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/Config.cmake.in)
+    install(TARGETS ${pkgName} DESTINATION ${CMAKE_INSTALL_PREFIX})
+else()
+    include(GNUInstallDirs)
+    install(TARGETS ${pkgName}
+            EXPORT ${pkgName}Targets
+            )
+
+    foreach(HDR_FILE ${HDRS})
+        get_filename_component(HDR_DIRECTORY ${HDR_FILE} DIRECTORY)
+        get_filename_component(ABSOLUTE_HDR_DIRECTORY ${HDR_DIRECTORY} ABSOLUTE)
+        file(RELATIVE_PATH RELATIVE_HDR_PATH ${CMAKE_CURRENT_SOURCE_DIR} ${ABSOLUTE_HDR_DIRECTORY})
+        install(FILES ${HDR_FILE} DESTINATION include/${RELATIVE_HDR_PATH})
+    endforeach()
+
+    include(CMakePackageConfigHelpers)
+    write_basic_package_version_file(
+            "${CMAKE_CURRENT_BINARY_DIR}/${pkgName}/${pkgName}ConfigVersion.cmake"
+            VERSION 0.0.1
```

Comment:
Is this version (0.0.1) for this project ?
How to decide/update this version (For now it is hard-coded) ?

Reply:
Yes it is. I think that I'll set its value in some variable and make it configurable in PreTarget script. What do you think?

Reply:
Yes. it's better to make this value configurable. For the sample (without a PreTarget), it can be 0.0.1 by default.

Kubernetes C client is planning version now. Thank you for your contribution !

Reply:
Done